pkg/controller/core/workload_controller.go

```diff
@@ -158,13 +158,27 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	case cancellingAdmission:
 		err := workload.UpdateStatusIfChanged(ctx, r.client, &wl, kueue.WorkloadAdmitted, metav1.ConditionFalse,
 			"AdmissionCancelled", "Admission cancelled")
+		if err != nil {
+			return ctrl.Result{}, client.IgnoreNotFound(err)
+		}
+
+		if err = workload.UpdateStatusIfChanged(ctx, r.client, &wl, kueue.WorkloadEvicted, metav1.ConditionTrue,
```

Comment:
We should update both in the same request. Otherwise there will be three API updates.

Furthermore, we will want to do this along with the eviction #603